src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -865,6 +873,47 @@ nubBy :: -- (IsStream t, Monad m) =>
     (a -> a -> Bool) -> t m a -> t m a
 nubBy = undefined -- fromStreamD . D.nubBy . toStreamD
 
+{-# INLINE nubByMerge #-}
+nubByMerge :: (IsStream t, Monad m) =>
+    (a -> a -> Bool) -> t m a -> t m a
+nubByMerge eq = fromStreamD . D.nubByMerge eq . toStreamD
+
+{-# INLINE removeDupsRight #-}
+removeDupsRight :: (IsStream t, MonadIO m) =>
+    (a -> a -> Ordering) -> t m a -> t m a -> t m a
+removeDupsRight eq s1 = fromStreamD . D.removeDupsAll eq (toStreamD s1) . toStreamD
+
+{-# INLINE differenceByMerge #-}
+differenceByMerge :: (IsStream t, MonadIO m) =>
+    (a -> a -> Ordering) -> t m a -> t m a -> t m a 
+differenceByMerge eq s1 = fromStreamD . D.mergeDifferenceJoinBy eq (toStreamD s1) . toStreamD
+
+{-# INLINE innerJoinByMerge #-}
+innerJoinByMerge :: (IsStream t, MonadIO m,  Eq a, Eq b) =>
+    (a -> b -> Ordering) -> t m a -> t m b -> t m (a, b)
+innerJoinByMerge eq s1 = fromStreamD . D.mergeByInner eq (toStreamD s1) . toStreamD
+
+{-# INLINE leftJoinByMerge #-}
+leftJoinByMerge :: (IsStream t, MonadIO m,  Eq a, Eq b) =>
+    (a -> b -> Ordering) -> t m a -> t m b -> t m (a, Maybe b)
+leftJoinByMerge eq s1 = fromStreamD . D.mergeByLeftInner eq (toStreamD s1) . toStreamD
+
+{-# INLINE rightJoinByMerge #-}
+rightJoinByMerge :: (IsStream t, MonadIO m,  Eq a, Eq b) =>
+    (a -> b -> Ordering) -> t m b -> t m a -> t m (Maybe b, a)
+rightJoinByMerge eq s1 = fromStreamD . D.mergeByRightInner eq (toStreamD s1) . toStreamD
+
+outerJoinByMerge :: (IsStream t, MonadIO m,  Eq a, Eq b)  =>
+    (a -> b -> Ordering) -> t m a -> t m b -> t m (Maybe a, Maybe b)
+outerJoinByMerge eq s1 = fromStreamD . D.mergeOuterJoin eq (toStreamD s1) . toStreamD    
+
+--mergeByIntersect
+intersectByMerge :: (IsStream t, MonadIO m, Eq a) =>
+    (a -> a -> Ordering) -> t m a -> t m a -> t m a
+intersectByMerge eq s1 = fromStreamD . D.mergeByIntersect eq (toStreamD s1) . toStreamD 
```

Comment:
Why add all these definitions in Transform module and then use them in the Top module? Why not directly define in the Top module? This seems like additional boilerplate code without any use.